src/status_im/ui/screens/activity_center/views.cljs

```diff
@@ -15,98 +15,114 @@
             [status-im.utils.handlers :refer [<sub >evt]]
             [quo.components.safe-area :as safe-area]))
 
-(defn activity-title
-  [{:keys [type]}]
-  (case type
-    constants/activity-center-notification-type-contact-request
-    (i18n/label :t/contact-request)
+;;;; Misc
 
-    constants/activity-center-notification-type-one-to-one-chat
-    "Dummy 1:1 chat title"
+(defn sender-name
+  [contact]
+  (or (get-in contact [:names :nickname])
+      (get-in contact [:names :three-words-name])))
 
-    "Dummy default title"))
+(defmulti notification-component :type)
 
-(defn activity-icon
-  [{:keys [type]}]
-  (case type
-    constants/activity-center-notification-type-contact-request
-    :main-icons2/add-user
-    :main-icons2/placeholder))
+;;;; Contact request notifications
 
-(defn activity-context
-  [{:keys [message last-message type]}]
-  (case type
-    constants/activity-center-notification-type-contact-request
-    (let [message     (or message last-message)
-          contact     (<sub [:contacts/contact-by-identity (:from message)])
-          sender-name (or (get-in contact [:names :nickname])
-                          (get-in contact [:names :three-words-name]))]
-      [[context-tags/user-avatar-tag
-        {:color          :purple
-         :override-theme :dark
-         :size           :small
-         :style          {:background-color colors/white-opa-10}
-         :text-style     {:color colors/white}}
-        sender-name
-        (multiaccounts/displayed-photo contact)]
-       [rn/text {:style {:color colors/white}}
-        (i18n/label :t/contact-request-sent)]])
-    nil))
+(defmethod notification-component constants/activity-center-notification-type-contact-request
+  [{:keys [id] :as notification}]
+  (let [message   (or (:message notification) (:last-message notification))
+        contact   (<sub [:contacts/contact-by-identity (:author notification)])
+        pressable (case (:contact-request-state message)
```

Comment:
Can we use `if since we have only two possible options?

Reply:
After thinking about it for a bit .. I think the current approach is better, Since it's easier to extend with other values.

Reply:
That's the idea @ibrkhalil, in theory we'll have different pressable alternatives, but maybe not. I'll keep an eye on this and simplify. Thanks for the feedback, YAGNI is a very important concept.